spec/support/shared_examples/collection_methods.rb

```diff
@@ -1,22 +1,21 @@
 shared_examples_for "Collection Methods" do
   it ".all returns a collection of objects" do
-    expect(api).to receive(:get).and_return(instance_double("Faraday::Result", :body => raw_collection.to_json))
+    expect(collection).to receive(:find_all_by_url).with(described_class.endpoint)
 
-    obj_collection = collection.all
-    expect(obj_collection).to        be_a Array
-    expect(obj_collection.length).to eq(2)
-    expect(obj_collection.first).to  be_a described_class
+    collection.all
```

Comment:
Should we call `.map` on `collection.all` and re add `expect(collection.length).to eq(2)` etc to better match the test name?  Or just change the name to reflect that an `Enumerator` is returned?